openapi3gen/openapi3gen_test.go

```diff
@@ -39,3 +39,17 @@ func TestExportedNonTagged(t *testing.T) {
 			"even_a_yaml": {Value: &openapi3.Schema{Type: "string"}},
 		}}}, schemaRef)
 }
+
+func TestExportUint(t *testing.T) {
+	type UnsignedIntStruct struct {
+		UnsignedInt uint `json:"uint"`
+	}
+
+	schemaRef, _, err := NewSchemaRefForValue(&UnsignedIntStruct{}, UseAllExportedFields())
+	require.NoError(t, err)
+	require.Equal(t, &openapi3.SchemaRef{Value: &openapi3.Schema{
+		Type: "object",
+		Properties: map[string]*openapi3.SchemaRef{
+			"uint": {Value: &openapi3.Schema{Type: "integer"}},
```

Comment:
More then `type: integer` this should probably also contain `minimum: 0`.
See the other cases such as
```go
	case reflect.Uint16:
		schema.Type = "integer"
		schema.Min = &zeroInt
		schema.Max = &maxUint16
```